sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1931,18 +1931,29 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         sql(
           """
             |SELECT c1, s, s * 10 FROM (
-            |  SELECT c1, (SELECT FIRST(c2) FROM t2 WHERE t1.c1 = t2.c1) s FROM t1)
+            |  SELECT c1, (SELECT MIN(c2) FROM t2 WHERE t1.c1 = t2.c1) s FROM t1)
             |""".stripMargin),
         correctAnswer)
       checkAnswer(
         sql(
           """
             |SELECT c1, s, s * 10 FROM (
-            |  SELECT c1, SUM((SELECT FIRST(c2) FROM t2 WHERE t1.c1 = t2.c1)) s
```

Comment:
Does this query also fail in other databases like pgsql?

Reply:
Isn't this subquery semantically the same as `SELECT c1, SUM((SELECT c2 FROM t2 WHERE t1.c1 = t2.c1 LIMIT 1)) s FROM t1 GROUP BY c1`? Spark currently does not support `LIMIT` to be on the correlation path, but this subquery, according to the current logic, is deterministic.

Reply:
> this subquery, according to the current logic, is deterministic

It seems fine to mark first/last deterministic? https://github.com/apache/spark/pull/34470#discussion_r742194450

Reply:
#29810 has been merged, @somani can you restore the original test?

Reply:
Done. Thanks @cloud-fan!